core/src/main/scala/org/elasticmq/actor/queue/operations/CommonOperations.scala

```diff
@@ -0,0 +1,31 @@
+package org.elasticmq.actor.queue.operations
+
+import org.elasticmq._
+import org.elasticmq.actor.queue.InternalMessage
+import org.elasticmq.util.NowProvider
+
+object CommonOperations {
+
+  def wasRegistered(
+      newMessage: NewMessageData,
+      deduplicationIdsHistory: FifoDeduplicationIdsHistory
+  ): Option[InternalMessage] = {
```

Comment:
is this ever used (the `InternalMessage`)? maybe it would be sufficient to store a `Set[DeduplicationId]`, instead of a map to messages?

Reply:
It is used in `SendMessage` action. If we find that the message is duplicated, we want to return the old one (with old message ID etc.)

Reply:
Ah ok, missed that :)